### DISCUSSION TITLE
Icon panel with summary feature

### DISCUSSION THREAD

**Discussion by guoranjun2** [freeplane/freeplane#1199] [Ideas]
Freeplane's filters are powerful and flexible, but unfortunately it doesn't have a clear view to summarize some of the node information we need, and the reason we need an icon summary function may be that it can help us quickly mark and jump to relevant nodes.
I have a lot of mind maps with thousands or even tens of thousands of nodes. I usually use scripts to filter the nodes, find the relevant nodes and then cancel the filter. This experience is not perfect. If there is an icon window to help me manage these nodes, the user experience will be much better.

In addition, I think mindmanager has done a good job at this. Here is my simulation using the mindmanager module combined with freeplane.

![image](https://github.com/freeplane/freeplane/assets/50692082/4acd1283-98a7-43f0-9215-e5a4eb61bf0b)

Do you think it is a good idea?

**Comment by sifran-github** [comment]
> If there is an icon window to help me manage these nodes, the user experience will be much better.

what's wrong with the filter by icon feature?
![image](https://github.com/freeplane/freeplane/assets/5730355/db336fe9-7cad-4b00-a9a1-8abad46019fe)

choose the icon and click on the "Quick filter" button.

**Reply by guoranjun2** [reply]
Before using the filter, you have no idea what results you will get, while the node information in the icon panel are displayed in real-time.

**Reply by sifran-github** [reply]
what information? just the count of how many nodes have this icon?

**Reply by guoranjun2** [reply]
> what information? just the count of how many nodes have this icon?
What I want is to display the title of the matching nodes under the icon, so that we can know what the result I want before using the filter
![image](https://github.com/freeplane/freeplane/assets/50692082/532d5020-7120-4efd-98c6-ac1d222acaa2)

**Reply by sifran-github** [reply]
I see.
Yeah, I don't think that Freeplane has this feature.

I will just add that if this feature is ever developed, I would want to be able to group not just by icon, but also by attribute, also specific attribute values (group by the values of a specific attribute), and maybe group by other features like style or custom script.

**Reply by guoranjun2** [reply]
> I see. Yeah, I don't think that Freeplane has this feature.
> 
> I will just add that if this feature is ever developed, I would want to be able to group not just by icon, but also by attribute, also specific attribute values (group by the values of a specific attribute), and maybe group by other features like style or custom script.

that would be fantastic.

**Reply by dpolivaev** [reply]
Now I see what you mean.

**Reply by dpolivaev** [reply]
Are you actually interested in all icons or in just some specific icons?
For example, I can imagine an implementation where you could define some filter conditions so that counts and lists of matching nodes could be available somewhere.

**Reply by guoranjun2** [reply]
I'm not sure if having too many icons would highly affect performance. If it does, we can consider using default icon set and extension scripts. My idea is to define a set of commonly used icons as default options, such as progress, priority, bookmarks, etc. If users have higher demands, they can use extension scripts to define filtering conditions. 
What do you think?

**Reply by dpolivaev** [reply]
Making it icon centered is one idea, making it filter condition centered is another idea.
If a certain set of relevant filter conditions e.g. icons was defined somewhere it could be shared and used somewhere else.

**Reply by guoranjun2** [reply]
Using scripts for filtering sounds more flexible and has stronger reusability. How about its performance compared to the previous idea? Will it still work normally if my mindmap has tens of thousands of nodes?

**Reply by dpolivaev** [reply]
Why are you mentioning scripts? Most filter conditions do not require scripting.
By the way, formula results and filtering script results are cached so that they are recalculated only if Freeplane assumes that the result could have changed.

**Reply by guoranjun2** [reply]
oh, I see.  I have been accustomed to using scripts for filtering, that's why I mentioned it inadvertently. so, it seems that using a filter would be a good choice.

**Reply by sifran-github** [reply]
@dpolivaev if you develop such thing, let me know. I have many ideas for it. I actually developed something similar for myself using scripts, but its just a "hack" for myself so i would not recommend using it.

one feature for example that i would like to see: the ability to drag and drop from this menu. for example, dropping a node here will change its priority to 4:
![image](https://github.com/freeplane/freeplane/assets/5730355/47b7d270-db1e-4793-8b40-b191e1ef5106)

its a bit complicated because it can be grouped by many options as i mentioned (icons, attributes, attributes values and more) and also because nodes can have multiple icons....

but.... with a smart implementation it can be a useful feature.

**Reply by dpolivaev** [reply]
Well, this idea does not work well with the idea of using filter conditions.
